pkgs/by-name/zi/zinit/package.nix

```diff
@@ -1,48 +1,70 @@
-{ stdenvNoCC, lib, fetchFromGitHub, installShellFiles }:
+{
+  stdenvNoCC,
+  lib,
+  fetchFromGitHub,
+  installShellFiles,
+  nix-update-script,
+}:
 
-stdenvNoCC.mkDerivation rec {
+stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "zinit";
   version = "3.13.1";
+
   src = fetchFromGitHub {
     owner = "zdharma-continuum";
-    repo = pname;
-    rev = "v${version}";
+    repo = finalAttrs.pname;
+    rev = "refs/tags/v${finalAttrs.version}";
     hash = "sha256-fnBV0LmC/wJm0pOITJ1mhiBqsg2F8AQJWvn0p/Bgo5Q=";
   };
-  # adapted from https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=zsh-zplugin-git
-  dontBuild = true;
+
+  outputs = [
+    "out"
+    "man"
+  ];
+
   strictDeps = true;
-  nativeBuildInputs = [ installShellFiles ];
-  installPhase = ''
-    outdir="$out/share/$pname"
 
-    cd "$src"
+  nativeBuildInputs = [ installShellFiles ];
 
-    # Zplugin's source files
-    install -dm0755 "$outdir"
-    # Installing backward compatibility layer
-    install -m0644 zinit{,-side,-install,-autoload}.zsh "$outdir"
-    install -m0755 share/git-process-output.zsh "$outdir"
+  installPhase = ''
+    runHook preInstall
 
-    installManPage doc/zinit.1
+    # Source files
+    mkdir -p $out/share/zinit
+    install -m0644 zinit{,-side,-install,-autoload}.zsh _zinit $out/share/zinit
+    install -m0755 share/git-process-output.zsh $out/share/zinit
 
-    # Zplugin autocompletion
+    # Autocompletion
     installShellCompletion --zsh _zinit
 
-    #TODO:Zplugin-module files
-    # find zmodules/ -type d -exec install -dm 755 "{}" "$outdir/{}" \;
-    # find zmodules/ -type f -exec install -m 744 "{}" "$outdir/{}" \;
+    # Manpage
+    mkdir -p ${placeholder "man"}/share/man/man{1..9}
```

Comment:
```suggestion
```

(Not needed)?

Reply:
https://github.com/zdharma-continuum/zinit/blob/c3469496455ea8b8ba184426c558e54817825bef/zinit.zsh#L1320

To prevent Zinit from trying to create directories in the read-only `/nix/store` path.